test/python/pulse/compiler_passes/test_schedule.py

```diff
@@ -0,0 +1,422 @@
+from qiskit.pulse.transforms import (
+    AlignLeft,
+    AlignRight,
+    AlignSequential,
+    AlignEquispaced,
+    AlignFunc,
+)
+from qiskit.pulse.compiler import MapMixedFrame, SetSequence, SchedulePass
+from qiskit.pulse.exceptions import PulseCompilerError
+from .utils import PulseIrTranspiler
+
+
+class SchedulingTestCase(QiskitTestCase):
+    """Base class for scheduling tests"""
+
+    def setUp(self):
+        super().setUp()
+        self._pm = PulseIrTranspiler([MapMixedFrame(), SetSequence(), SchedulePass()])
+
+
+class TestScheduleAlignLeft(SchedulingTestCase):
+    """Test Schedule pass with Left Alignment"""
+
+    def test_single_instruction(self):
+        """test with a single instruction"""
+
+        ir_example = SequenceIR(AlignLeft())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = self._pm.run(ir_example)
+        self.assertEqual(ir_example.initial_time(), 0)
+        self.assertEqual(ir_example.final_time(), 100)
+
+    def test_bad_or_missing_sequencing(self):
+        """test that bad or missing sequencing raises an error"""
+
+        ir_example = SequenceIR(AlignLeft())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        with self.assertRaises(PulseCompilerError):
+            SchedulePass().run(ir_example)
+
+        ir_example.sequence.add_edges_from_no_data([(1, 0), (2, 0)])
+        with self.assertRaises(PulseCompilerError):
+            SchedulePass().run(ir_example)
+
+    def test_parallel_instructions(self):
+        """test with two parallel instructions"""
+
+        ir_example = SequenceIR(AlignLeft())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(200, 0.1), mixed_frame=MixedFrame(Qubit(1), QubitFrame(1))))
+
+        ir_example = self._pm.run(ir_example)
+        self.assertEqual(ir_example.initial_time(), 0)
+        self.assertEqual(ir_example.final_time(), 200)
+        self.assertEqual(ir_example.scheduled_elements()[0][0], 0)
+        self.assertEqual(ir_example.scheduled_elements()[1][0], 0)
+
+    def test_sequential_instructions(self):
+        """test with two sequential instructions"""
+
+        ir_example = SequenceIR(AlignLeft())
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = self._pm.run(ir_example)
+        self.assertEqual(ir_example.initial_time(), 0)
+        self.assertEqual(ir_example.final_time(), 200)
+        self.assertEqual(ir_example.scheduled_elements()[0][0], 0)
+        self.assertEqual(ir_example.scheduled_elements()[1][0], 100)
+
+    def test_multiple_children(self):
+        """test for a graph where one node has several children"""
```

Comment:
Maybe the name is successor and predecessor? Qiskit often use these words for position in the graph. children/parents are also used for the nested relationship in ScheduleBlock, and it confused me a bit.